src/SetupApi/SetupApi.cs

```diff
@@ -219,6 +219,272 @@ public static partial class SetupApi
+        /// A <see cref="OpenFlags"/> that controls how the device information element is opened.
+        /// </param>
+        /// <param name="openFlags">
+        /// A variable of <see cref="OpenFlags"/> that controls how the device information element is opened.
+        /// </param>
+        /// <param name="deviceInfoData">
+        /// A pointer to a caller-supplied <see cref="SP_DEVINFO_DATA"/> structure that receives information about the device information
+        /// element for the device instance that is specified by <paramref name="deviceInstanceId"/>. The caller must set <see cref="SP_DEVINFO_DATA.Size"/>.
+        /// This parameter is optional and can be <see langword="null"/>.
+        /// </param>
+        /// <returns>
+        /// The function returns <see langword="true"/> if it is successful.
+        /// Otherwise, it returns <see langword="false"/> and the logged error can be retrieved with a call to <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        /// <see href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff552071(v=vs.85).aspx"/>
+        [DllImport(nameof(SetupApi), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SetupDiOpenDeviceInfo(
+            SafeDeviceInfoSetHandle deviceInfoSet,
+            string deviceInstanceId,
+            IntPtr parent,
+            OpenFlags openFlags,
+            ref SP_DEVINFO_DATA deviceInfoData);
+
+        /// <summary>
+        /// The <see cref="SetupDiSetSelectedDevice"/> function sets a device information element as the selected member of a device information set.
+        /// This function is typically used by an installation wizard.
+        /// </summary>
+        /// <param name="deviceInfoSet">
+        /// A handle to the device information set that contains the device information element to set as the selected member of the device information set.
+        /// </param>
+        /// <param name="deviceInfoData">
+        /// A <see cref="SP_DEVINFO_DATA"/> structure that specifies the device information element in <paramref name="deviceInfoSet"/> to set as the
+        /// selected member of <paramref name="deviceInfoSet"/>.
+        /// </param>
+        /// <returns>
+        /// The function returns <see langword="true"/> if it is successful.
+        /// Otherwise, it returns <see langword="false"/> and the logged error can be retrieved with a call to <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff552176(v=vs.85).aspx"/>
+        [DllImport(nameof(SetupApi), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SetupDiSetSelectedDevice(
+            SafeDeviceInfoSetHandle deviceInfoSet,
+            ref SP_DEVINFO_DATA deviceInfoData);
```

Comment:
Please type the second parameter as:

```cs
[Friendly(FriendlyFlags.In)] SP_DEVINFO_DATA*
```

This way you'll get a friendly overload that doesn't even require `ref`, as well as offering the caller the maximum flexibility as far as where the data is for the struct.